### PR TITLE
SMAA Support in HDRP Postprocess stack

### PR DESCRIPTION
**Pull request by FrancescoC-unity**
### Purpose of this PR
Adding SMAA as an antialiasing option. 

---
### Release Notes
Added SMAA support. 

---
### Testing status
**Katana Tests**: https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders?ScriptableRenderLoop_branch=HDRP%2FSMAA&automation-tools_branch=master&unity_branch=trunk# [Just kicked, ignore the failing one]

**Manual Tests**: What did you do?
- [ ] Opened test project + Run graphic tests locally
- [x] Built a player
- [x] Checked new UI names with UX convention
- [ ] Tested UI multi-edition + Undo/Redo
- [x] C# and shader warnings (supress shader cache to see them)
- Other: 

---
### Overall Product Risks
**Technical Risk**: Low

**Halo Effect**: Low

---
### Comments to reviewers

Two notes: 

1- I used SMAA all caps (as it is an acronym) even to start the name of the variables as it is the actual name (smaa is technically incorrect naming), but really I can change that if it feels against the conventions. 

2- The stencil part is slightly awkward, I have already in mind a change to make that cleaner, but I want to do it in a separate PR as it will touch files completely unrelated and is a more general change. I will do it straight after this is merged. 

I have an idea for an optimization that allow moving to compute, but it won't save much, so I am holding off since it is quite a substantial change.

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Runtime/PostProcessing/Shaders/SubpixelMorphologicalAntialiasing.shader [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+#pragma only_renderers d3d11 ps4 xboxone vulkan metal switch
+
+        ENDHLSL
+
+        SubShader
+    {
+        Cull Off ZWrite Off ZTest Always
+
+        // 0 - Edge detection (Low)
```
Comment (Chman) [inline com.unity.render-pipelines.high-definition/Runtime/PostProcessing/Shaders/SubpixelMorphologicalAntialiasing.shader]: Move to `multi_compile_local` on `SMAA_PRESET_X` to simplify the code.

File: com.unity.render-pipelines.high-definition/Runtime/PostProcessing/Shaders/SubpixelMorphologicalAntialiasing.shader [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+                #pragma vertex VertEdge
+                #pragma fragment FragEdge
+                #define SMAA_PRESET_HIGH
+                #include "SubpixelMorphologicalAntialiasingBridge.hlsl"
+
+            ENDHLSL
+        }
+
+        // 3 - Blend Weights Calculation (Low)
```
Comment (Chman) [inline com.unity.render-pipelines.high-definition/Runtime/PostProcessing/Shaders/SubpixelMorphologicalAntialiasing.shader]: Same, `multi_compile_local`

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Camera/HDCameraUI.Drawers.cs [hunk 3]
```diff
@@ -425,6 +425,10 @@ static void Drawer_FieldClear(SerializedHDCamera p, Editor owner)
         static void Drawer_Antialiasing(SerializedHDCamera p, Editor owner)
         {
             p.antialiasing.intValue = EditorGUILayout.Popup(antialiasingContent, p.antialiasing.intValue, antialiasingModeNames);
+            if(p.antialiasing.intValue == (int)HDAdditionalCameraData.AntialiasingMode.SubpixelMorphologicalAntiAliasing)
+            {
+                p.SMAAQuality.intValue = EditorGUILayout.Popup(SMAAQualityPresetContent, p.SMAAQuality.intValue, SMAAQualityNames);
```
Comment (Chman) [inline com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Camera/HDCameraUI.Drawers.cs]: Can't we just use `EditorGUILayout.PropertyField` and remove the `SMAAQualityNames` stuff from `HDCameraUI.Skin.cs` to simplify things? The `SMAAQualityLevel` enum already has properly formatted names, so...

Comment (FrancescoC-unity) [reply com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Camera/HDCameraUI.Drawers.cs]: Ha, yes of course, I was unaware of the magicness behind that.

**Comment by FrancescoC-unity** [comment]
Note that katana tests pass, it is only one LWRP test that fails.

**Pull request merged**